### ISSUE TITLE
Error "Failed to get logs for block" on eth_getCalls 

### ISSUE THREAD

**Issue by zimbabao** [ethereum/go-ethereum#24606]
#### System information

Geth version: `1.10.13-stable`
OS & Version: linux

```
curl http://localhost:8545 --data '{"jsonrpc":"2.0","id":0,"method":"eth_getLogs","params":[{"fromBlock":"0xbf2084","address":"0xC12BA48c781F6e392B49Db2E25Cd0c28cD77531A","topics":["0x4ac0014773275a3dfb58c58539631006301de41998cce7c4f8698d297c88bb2d"]}]}' -H 'content-type:application/json'
```

#### Expected behaviour

```
curl HTTP://localhost:8545 --data '{"jsonrpc":"2.0","id":0,"method":"eth_getLogs","params":[{"fromBlock":"0xbf2084","address":"0xC12BA48c781F6e392B49Db2E25Cd0c28cD77531A","topics":["0x4ac0014773275a3dfb58c58539631006301de41998cce7c4f8698d297c88bb2d"]}]}' -H 'content-type:application/json'

{"jsonrpc":"2.0","id":0,"result":[{"address":"0xc12ba48c781f6e392b49db2e25cd0c28cd77531a","blockHash":"0x4519d55cb703ff0e8606625d476a25ab382dd7fabb0e01c1a314c3fcd635c8ac","blockNumber":"0xbf2084","data":"0x541ad634cce9293e20f16e25e00ae7e4d003bfae2a9e4d24bc1c1d2db448dce6","logIndex":"0x4e","removed":false,"topics":["0x4ac0014773275a3dfb58c58539631006301de41998cce7c4f8698d297c88bb2d"],"transactionHash":"0xd71d3c90adcce0fabc903fceed668561c92f5be5d8837295f9e46e2f6d99894e","transactionIndex":"0x1a"}]}rajaram@:/Users/rajaram$
```
#### Actual behaviour

```
time curl http://localhost:8545 --data '{"jsonrpc":"2.0","id":0,"method":"eth_getLogs","params":[{"fromBlock":"0xbf2084","address":"0xC12BA48c781F6e392B49Db2E25Cd0c28cD77531A","topics":["0x4ac0014773275a3dfb58c58539631006301de41998cce7c4f8698d297c88bb2d"]}]}' -H 'content-type:application/json'
curl: (52) Empty reply from server

real	6m21.627s
user	0m0.011s
sys	0m0.025s

and logs shows
WARN [03-30|00:03:29.945] Served eth_getLogs                       conn=127.0.0.1:54702 reqid=0      t=6m20.87751607s err="failed to get logs for block"
```

#### Steps to reproduce the behaviour
1. Make request for logs 
```
curl HTTP://localhost:8545 --data '{"jsonrpc":"2.0","id":0,"method":"eth_getLogs","params":[{"fromBlock":"0xbf2084","address":"0xC12BA48c781F6e392B49Db2E25Cd0c28cD77531A","topics":["0x4ac0014773275a3dfb58c58539631006301de41998cce7c4f8698d297c88bb2d"]}]}' -H 'content-type:application/json'
```
2. Check logs

#### Backtrace

````
[backtrace]
````

This issue is similar to #23875 
When submitting logs: please submit them as text and not screenshots.

**Comment by s1na** [comment]
Did you notice any other warnings or errors around the log line (`WARN [03-30|00:03:29.945] Served eth_getLogs`) that you posted?

**Comment by holiman** [comment]
Related to the empty response: https://github.com/golang/go/issues/47229 

When we hit a timeout, the response becomes empty.

**Comment by holiman** [comment]
The request is to gather all logs from block `12525700` and onwards, which is going to take a while ....

**Comment by holiman** [comment]
Please try the latest release, and also provide some logs from the geth startup sequence

**Comment by ibadullaev-inc4** [comment]
same questions
```
[nariman@notebook chainscale]$ curl --data '{"jsonrpc": "2.0", "id": 1, "method":"eth_getLogs", "params": [{"address":["0x0de2669e8a7a6f6cc0cbd3cf2d1eead89e243208"],"fromBlock":"0x0","toBlock":"latest","topics":[["0xe15729a2f427aa4572dab35eb692c902fcbce57d41642013259c741380809ae2"],["0x0000000000000000000000000000000000000000000000000000000000000001"]]}]}' -H "Content-Type: application/json" -X POST http://127.0.0.1:8545
curl: (52) Empty reply from server

```

**Comment by dvbfcm850** [comment]
same questions

**Comment by s1na** [comment]
> "fromBlock":"0x0","toBlock":"latest"

I bet this request is taking longer than 30s and is timing out. I'm working on returning an error in case of timeout so it's less confusing: https://github.com/ethereum/go-ethereum/pull/25457

**Comment by s1na** [comment]
Fixed via #25457
